app/src/main/java/com/startup/go4lunch/ui/MapFragmentViewModel.java

```diff
@@ -5,48 +5,139 @@
+        if (string != null) {
+            Restaurant restaurant = getRestaurantFromString(string);
+            if (restaurant != null) {
+                Location location =new Location(restaurant.getName());
+                location.setLatitude(restaurant.getLatitude());
+                location.setLongitude(restaurant.getLongitude());
+                mapCenterLocationLiveData.setValue(location);
+            }
+        }
+    };
+
     @NonNull
-    public LiveData<String> getSearchStringLivedata() {
-        return searchRepository.getMapFragmentSearchLivedata();
+    private LiveData<Restaurant> getRestaurantToCenterOnLiveData() {
+        return Transformations.map(searchRepository.getMapFragmentSearchLivedata(), string -> {
+            if (string != null) {
+                return restaurantRepository.getRestaurantResearchedFromString(string);
+            }
+            return null;
+        });
+    }
+
+    private final Observer<List<Restaurant>> restaurantListObserver = restaurantList -> {
+        List<RestaurantMapMarker> restaurantMapMarkerList = new ArrayList<>();
+        if (restaurantList != null) {
+            for (Restaurant restaurant : restaurantList) {
+                restaurantMapMarkerList.add(new RestaurantMapMarker(restaurant, workmateLunchOnRestaurant(restaurant.getId())));
+            }
+        }
+        restaurantMapMarkerListLiveData.setValue(restaurantMapMarkerList);
+    };
+
+    private final Observer<List<Workmate>> workmateListObserver = workmatesList -> {
+        List<RestaurantMapMarker> restaurantMapMarkerList = restaurantMapMarkerListLiveData.getValue();
+        if (restaurantMapMarkerList != null) {
+            for(RestaurantMapMarker restaurantMapMarker: restaurantMapMarkerList) {
+                restaurantMapMarker.setWorkmateLunchOnRestaurant(workmateLunchOnRestaurant(restaurantMapMarker.getRestaurant().getId()));
+            }
+        }
+        restaurantMapMarkerListLiveData.setValue(restaurantMapMarkerList);
+    };
+
+    public void setEndSearch() {
+        searchRepository.setMapFragmentSearch(null);
     }
 
     @Nullable
-    public Restaurant getRestaurantFromString(@NonNull String string) {
+    private Restaurant getRestaurantFromString(@NonNull String string) {
         return restaurantRepository.getRestaurantResearchedFromString(string);
     }
 
-    public void setEndSearch() {
-        searchRepository.setMapFragmentSearch(null);
+    private boolean workmateLunchOnRestaurant(long restaurantUid) {
+        List<Workmate> workmateList = workmateRepository.getWorkmateListLiveData().getValue();
+        if (workmateList != null) {
+            for (Workmate workmate : workmateList) {
+                if (workmate.getRestaurantSelectedUid() != null) {
+                    if (workmate.getRestaurantSelectedUid() == restaurantUid) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
```

Comment:
Ca me fait repenser à la remarque que tu m'as faite en pair programming la semaine dernière. Si tu veux limiter le nombre de conditions imbriqués, tu peux modifier comme cela en gérant les cas d'erreur au début de la méthode.

```suggestion
        if (workmateList == null) return false;
        if (restaurantUid == null) return false;
        
        for (Workmate workmate : workmateList) {         
            if (workmate.getRestaurantSelectedUid() == restaurantUid) {
                return true;
            }
        }
        
        return false;
```

Reply:
La on est obligé de tester comme même workmate.getRestaurantSelectedUid() != null a chaque itération, mais c'est une bonne pratique ;)